src/Activities/Internal/ActivityMetricsSender.cs

```diff
@@ -73,6 +81,16 @@ public void SendActivityMetric(Activity activity)
 				}
 			}
 
+			bool isSetParentNameAsDimensionEnabled = m_activityOptions.Value.SetParentNameAsDimension;
```

Comment:
In terms of performance, I believe that having m_SetParentNameAsDimensionEnabled as a class property would improve the performance, because we do not need to access IOption object > access value > access SetParentNameAsDimension object property. 
This will reduce 2 additional accessing steps.

Reply:
Differences in performance are negligible, IOptions.Value is created and cached at first access.